stubs/dj-database-url/dj_database_url.pyi

```diff
@@ -0,0 +1,19 @@
+from collections.abc import Mapping
+from typing_extensions import TypedDict
+
+DEFAULT_ENV: str
+SCHEMES: Mapping[str, str]
+
+class _DBConfig(TypedDict):
+    NAME: str
+    USER: str
+    PASSWORD: str
+    HOST: str
+    PORT: str
+    CONN_MAX_AGE: int
+    ENGINE: str
```

Comment:
Not all keys will _necessarily_ always be in the dictionary returned from `parse()` and `config`. In fact, if I'm reading the source code correctly, the only key that's _guaranteed_ to be present is `"ENGINE"`. Ideally we'd use [PEP 655](https://peps.python.org/pep-0655/) to express that here, but not all type checkers support PEP 655 yet, so the best we can do in typeshed is something like this. The "required" key is given in `_DBConfigBase`, and the "optional" keys are given in the subclass, `_DBConfig`, which is marked with the `total=False` keyword (see [the description of this trick in the `typing` docs](https://docs.python.org/3/library/typing.html#typing.TypedDict.__optional_keys__)).

You also missed the `"OPTIONS"` key (https://github.com/jazzband/dj-database-url/blob/836740dd814878cfa9957a0a68b545060ba89c10/dj_database_url.py#L157-L158) :)

```suggestion
class _DBConfigBase(TypedDict);
    NAME: str

class _DBConfig(_DBConfigBase, total=False):
    USER: str
    PASSWORD: str
    HOST: str
    PORT: str
    CONN_MAX_AGE: int
    OPTIONS: dict[str, Any]
    ENGINE: str
```

Reply:
Yeah, I think you're right on all points. I've used your suggested code to fix the issues pointed out.

Thanks for the review. I really appreciate it.
Also, thanks for letting me know about PEP 655 and the `total=False` trick.

Reply:
> Thanks for the review. I really appreciate it.

Pleasure! :D